### PR TITLE
DCS-1688 Showing warning when an arrival has more than 115 body scans…

### PR DESCRIPTION
**Pull request by andrewrlee**
… within the last year

### CODE REVIEW

File: server/@types/welcome/index.d.ts [hunk 1]
```diff
@@ -12,6 +12,13 @@ declare module 'welcome' {
   export type PotentialMatch = schemas['PotentialMatch']
   export type PrisonerDetails = schemas['PrisonerDetails']
 
+  export const enum LocationType {
+    COURT = 'COURT',
+    CUSTODY_SUITE = 'CUSTODY_SUITE',
+    PRISON = 'PRISON',
+    OTHER = 'OTHER',
+  }
```
Comment (andrewrlee) [inline server/@types/welcome/index.d.ts:20]: moved this from the arrival service

File: server/routes/temporaryabsences/temporaryAbsencesController.test.ts [hunk 2]
```diff
@@ -1,50 +1,24 @@
...
-    dateOfBirth: '1970-01-01',
-    prisonNumber: 'G0012HK',
-    reasonForAbsence: 'External visit',
-    movementDateTime: '2022-01-16T12:30:00',
-  },
+const temporaryAbsences: TemporaryAbsence[] = [
+  createTemporaryAbsence(),
+  createTemporaryAbsence(),
+  createTemporaryAbsence(),
+  createTemporaryAbsence(),
```
Comment (andrewrlee) [inline server/routes/temporaryabsences/temporaryAbsencesController.test.ts:21]: doesn't matter the details and the factory creates correctly structured versions

File: server/routes/temporaryabsences/checkTemporaryAbsenceController.test.ts [hunk 3]
```diff
@@ -41,65 +33,60 @@ afterEach(() => {
...
-      .get('/prisoners/G0013AB/check-temporary-absence')
+      .get('/prisoners/A1234AA/check-temporary-absence')
       .expect('Content-Type', 'text/html; charset=utf-8')
       .expect(res => {
         const $ = cheerio.load(res.text)
         expect($('h1').text()).toContain('This person will be returned to prison')
-        expect($('.data-qa-name').text()).toContain('John Doe')
-        expect($('.data-qa-dob').text()).toContain('1 January 1971')
-        expect($('.data-qa-prison-number').text()).toContain('G0013AB')
-        expect($('.data-qa-reason-for-absence').text()).toContain('Hospital appointment')
```
Comment (andrewrlee) [inline server/routes/temporaryabsences/checkTemporaryAbsenceController.test.ts:66]: we don't tend to test details and these are now incorrect as now using factory created version

File: server/routes/bookedtoday/transfers/checkTransferController.test.ts [hunk 4]
```diff
@@ -6,29 +6,20 @@ import { TransfersService, RaiseAnalyticsEvent } from '../../../services'
...
-  pncNumber: '01/5678A',
-  date: '2021-09-01',
-  fromLocation: 'Leeds',
-}
-
 beforeEach(() => {
   app = appWithAllRoutes({ services: { transfersService, raiseAnalyticsEvent }, roles: [Role.PRISON_RECEPTION] })
   config.confirmEnabled = true
-  transfersService.getTransfer.mockResolvedValue(transfer)
+  transfersService.getTransfer.mockResolvedValue(createTransfer())
```
Comment (andrewrlee) [inline server/routes/bookedtoday/transfers/checkTransferController.test.ts:20]: use factory version

**Review by davidthomas22 — APPROVED** [review]
LGTM

**Pull request merged**